api/provider/aws/environment_link.go

```diff
@@ -0,0 +1,71 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (e *EnvironmentProvider) Link(sourceEnvironmentID, destEnvironmentID string) error {
+	fqSourceEnvID := addLayer0Prefix(e.Config.Instance(), sourceEnvironmentID)
+	fqDestEnvID := addLayer0Prefix(e.Config.Instance(), destEnvironmentID)
+
+	sourceGroup, err := readSG(e.AWS.EC2, getEnvironmentSGName(fqSourceEnvID))
+	if err != nil {
+		return err
+	}
+
+	destGroup, err := readSG(e.AWS.EC2, getEnvironmentSGName(fqDestEnvID))
+	if err != nil {
+		return err
+	}
+
+	ingressInput := e.createIngressInput(sourceGroup.GroupId, destGroup.GroupId)
```

Comment:
The current pattern is wrapping all aws calls in helpers with more sensible param types. 
Please refer to the [rules of thumb page](https://github.com/quintilesims/layer0/blob/api-refactor/api/provider/aws/README.md)

 So this code would look something like:
```
sourceGroup, err := ...
destGroup, err := ...

sourceGroupID := aws.StringValue(sourceGroup.GroupId)
destGroupID := aws.StringValue(destGroup.GroupId)
if err := e.createIngressInput(sourceGroupID, destGroupID); err != nil { ... }

if err := e.createIngressInput(destrGroupID, sourceGroupID); err != nil { ... }
```

Reply:
✅